### PR TITLE
The White Ship's round-start dock is now random

### PR DESCRIPTION
**Pull request by coiax**
:cl: coiax
add: The White Ship now has equal chances of being at any valid dock
at roundstart. (The default docks are Deep Space, Station and Lavaland,
and if the Derelict ruin is selected, it may be docked there as well.)
/:cl:

Thanks to @MrStonedOne for his help.

EDIT:
**Reasoning**

1) This is going to be part of my plan to add shuttle beacons to activate on space ruins, thus allowing the white ship to dock at those locations. The white ship will have then to have its locations unlocked through manual travel.

2) The White Ship is also fun, and offers evacuation possibilities when it is at the station.

3) Blood cultists cannot make runes on the white ship anyway (although I disagree with this) so there's little balance consideration re:bloodcult.

### CODE REVIEW

**Comment by Kevak** [comment]
It also has a South Of Derelict dock if I recall but only if the derelict spawns.

**Comment by Iamgoofball** [comment]
dont forget docking at KSS13 if it's available

**Comment by coiax** [comment]
I did remember the derelict, but wanted to mention the guaranteed docks only. I'll edit the changelog.

File: code/__HELPERS/text.dm [hunk 1]
```diff
@@ -179,6 +179,7 @@
 
 //Checks the beginning of a string for a specified sub-string
 //Returns the position of the substring or 0 if it was not found
+//Also could be called `startswith`
```
Comment (Jordie0608) [inline code/__HELPERS/text.dm]: Is this really needed =P

**Comment by AdamElTablawy** [comment]
i dislike the idea of 'crashed, abandoned medical ship' randomly being at the station roundstart for numerous reasons

1. doesn't make much sense for an abandoned medical ship to just already be there

2. makes new players who want to find it sad because they just can, + it comes with a captain's laser equivalent

3. clock cult is gonna use this as a base instead of the hotel whenever possible and the only thing worse than an off station clockcult is an off station clockcult that can fly to lavaland at will for no component downsides

**Comment by Crushtoe** [comment]
Is it possible to set it to have weights; Deep Space is preferred over Lavaland which is preferred over already coming Docked?

**Comment by Mindustry** [comment]
@AdamElTablawy
Did you miss #22101?

**Comment by FantasticFwoosh** [comment]
@Mindustry i think they were referring more to bloodcult. Which currently feels no penalty.

> Which nobody has touched in like a millennia github wise given that we have 10 pr's a day the last few weeks or so

**Comment by Kevak** [comment]
Bloodcult is actually incapable of drawing runes on the white ship. Period. There's a proc that checks for it.

Bloodcult is also limited to station zlevel or lavaland for drawing runes. They are utterly incapable of drawing them off of those two zlevels. Unlike clockcult which can still do things offstation in a limited and far slower fashion.

**Comment by AnturK** [comment]
Making dock subtypes is all well and good but why the random dock

**Comment by KingPhilipIII** [comment]
Well. Let's be honest. What are the odds a cult spawns and the ship randomly docks. If that's a problem it won't be a common one anyway. The spess hotel is always around though. 

With regards to logic of an abandoned ship docking. Just say the ship floated by and it's auto docking procedures engaged. The emergency shuttle does it after all. 

I've seen people take chunks of the crew out to raid pirates too anyway. I remember being the ship's engineer. And then someone planted a Syndicate bomb and killed us all.

**Comment by WJohn** [comment]
Huh, didn't know you could have MULTIPLE start locations for ships, and it would pick one of them. That's really cool! I think I'll be using this for future white ship designs.

**Comment by Cheridan** [comment]
bloodcult will still want to take the white ship down to lavaland to build a base

**Comment by coiax** [comment]
So we're removing fun things from the game because of MUH CULT BALANCE?

**Comment by Cheridan** [comment]
Yes?? When you add new shit to the game you kind of have to be sure that it doesn't break existing game modes.

**Comment by ExcessiveUseOfCobblestone** [comment]
how is bringing the white ship home roundstart a fun feature to begin with when it takes one of the only good things about exploring space out of exploring space...?

**Comment by coiax** [comment]
I don't see the white ship anymore. It's sad.

Secondly, I'm going to be adding white ship docking ports to a variety of ruins, and they'd be able to take advantage of the randomness.

Also, as said before, bloodcult cannot make runes on shuttles. Period.

**Comment by RemieRichards** [comment]
Don't add docks to ruins.
it (sorry for lack of better wording) ruins the exploration part of it.

**Comment by coiax** [comment]
I was going to make the docks required to be activated first. So you'd have to visit the ruin, and then enable the docking system, allowing travel between unlocked locations via shuttle.

**Comment by RemieRichards** [comment]
That's fair, please specify something like that beforehand cos it's changed my opinion.

**Comment by optimumtact** [comment]
No consensus here.

**Comment by coiax** [comment]
This being merged is part of my long term plan for the White Ship to serve as an unlockable gateway to a variety of space ruins. With more ruins that have White Ship docks in the game, the chances of being on station will decease.

But I think having it start on station still gives some assistants a chance to become space pirates and go off and get up to mischief.

**Comment by jughu** [comment]
Yarr!

**Comment by KingPhilipIII** [comment]
I'm still in support of this. Not enough spess exploration happens. Maybe people will feel motivated with their new vessel.

**Comment by Crushtoe** [comment]
I'm in support of this. It makes the white ship a fun thing again that's more than just a pirate ship or roundstart EVA and teleporter trip. It can be used for fun gimmicks and enjoy actual usage if the idea of revisiting old ruins takes off.

Further, it means you have a reason to go check the derelict and other places and not just jetpack straight to the same place. It actually makes exploring multiple areas a thing.

**Comment by Cheridan** [comment]
This PR is fine as long as the white ship can't start on the station, it's just too much of a gamechanger for it to be a diceroll. 
**If it's there:** Congrats, you now have isolated vulnerable targets with no comms to convert or kill as you please. You are also away from Sec + AI + Maintwanderers who might stumble onto you. EZ game.
**If it's not:** Sucks to be you.

It would be better if it was there 100% of the time but that would be even worse, assistants have murdered each other over who gets yellow gloves much less who gets a goddamn space ship.

**Comment by coiax** [comment]
Yeah, but sometimes cults and traitors are lucky, see: Hop is a cultist/traitor/changeling. Sometimes the round is in the antagonist's favour?

**Comment by KorPhaeron** [comment]
White ship should start docked as the station and require multiple head approval to launch at the ID swipe things.

There should be a white ship dock on every z level

There should be a way to remotely disable or attack it so it doesn't turn into a flying cult fort

I think that'd be a fun way to add simple space exploration

**Comment by KingPhilipIII** [comment]
I like Kor's idea. Leave the white ship as it is and maybe add an NT vessel designed for spesss travel. Again with several locks to prevent cult forts and the like. If someone wants to go be a pirate space captain they can go hunt for the regular white ship or they can play by NT's rules.

**Comment by optimumtact** [comment]
Just add a console on the bridge so the heads can recall the white shuttle at any time and it should be reasonably balanced

**Comment by KingPhilipIII** [comment]
a console that any robust player could steal with a little effort and the tools conferred by their traitordom.

**Comment by optimumtact** [comment]
So make it reconstructable and printable at the circuit board printer.

**Comment by optimumtact** [comment]
Please do the needful with a console

**Comment by coiax** [comment]
Since people aren't going to merge this plainly, I will need to work on my "docking beacons" code. So this is WIP for now.

**Pull request closed without merging**